keps/sig-network/1864-disable-lb-node-ports/README.md

```diff
@@ -136,6 +151,169 @@ re-enabling node port should not cause any traffic disruptions.
 Version skew from the control plane to kube-proxy should be trivial since kube-proxy's behavior is driven by the `nodePort` field
 and not the `allocateLoadBalancerNodePorts` field.
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: ServiceLBNodePortControl
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+
+No, enabling the feature gate but not setting `spec.allocateLoadBalancerNodePorts` will not
+change any default behaviors in Service.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes, if the feature gate is disabled, new Services cannot use the new field, but existing Services
```

Comment:
If the feature gate is off, and a Service has the field set to `false` what happens on update?

Reply:
Updates are still allowed, assuming the field was set when the feature gate was enabled.

Reply:
I'll clarify that

Reply:
What should happen if:


* Enable gate
* Create a svc with `spec.allocateLoadBalancerNodePorts: off`
* Observe node port was NOT allocated
* Disable gate

Do we expect the service stack to allocate a node port immediately?  How about on the next update?  Do we just leave nodePort unallocated?

We'll need good tests around this

Reply:
ack